Identify taxiways for ground trajectories

RaphaelDELAIR opened the pull request:
the internal function get_long_twy should be moved to airports.py ?

xoolive commented on traffic/algorithms/navigation.py:
```diff
@@ -1102,3 +1104,135 @@ def slow_taxi(
             if candidate is not None:
                 yield candidate
+
+    def ground_movement_type(self, airport):
```
why not a nested function in `assign_twy()`?

xoolive commented:
@RaphaelDELAIR 

I edited your PR into two functions:
- [ ] `ground_trajectory` which does (at the moment, just a placeholder) a very basic job based on the airport footprint (you should probably improve it based on heuristics to determine before take-off or after landing, not based on the `origin` and `destination` labels please);
- [ ] `on_taxiway` returns a `FlightIterator`. See below two examples, the thresholds should probably be adjusted to fit your needs.

I think those examples could be used to write unit tests if you can.

```python
from traffic.data.samples import elal747

elal747.first("50T").map_leaflet(
    airport="LIRF",
    zoom=14,
    highlight=dict(red=lambda f: f.on_taxiway("LIRF", tolerance=35)),
)
```

<img width="971" alt="Capture d’écran 2021-07-13 à 23 38 51" src="https://user-images.githubusercontent.com/1360812/125528784-482e5862-abfc-4620-b862-6d6ea6f8e854.png">

```python
elal747.last("15T").map_leaflet(
    zoom=14,
    airport="LLBG",
    highlight=dict(red=lambda f: f.on_taxiway("LLBG", tolerance=35)),
)
```

<img width="966" alt="Capture d’écran 2021-07-13 à 23 39 32" src="https://user-images.githubusercontent.com/1360812/125528848-d66bc4ff-2e0e-4e58-a641-178f528d00d1.png">

- [ ] also please consider refactoring the idea to improve `Flight.on_parking_position()`

RaphaelDELAIR commented:
@xoolive I have updated those methods and added corresponding tests

However the last line of the test method need a flight present in a the pickle file attached to be added to 'zurich_airport' dataset. It is now commented to make the test pass but should be uncommented when this is done.

Sincerely,

Raphael

[zurichairport.zip](https://github.com/xoolive/traffic/files/6907877/zurichairport.zip)
